tests/ExampleSchema.hs

```diff
@@ -0,0 +1,304 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE DeriveGeneric #-}
+{-# LANGUAGE TypeOperators #-}
+{-# LANGUAGE ViewPatterns #-}
+
+-- | An example GraphQL schema, used in our end-to-end tests.
+--
+-- Based on the example schema given in the GraphQL spec. See
+-- <https://facebook.github.io/graphql/#sec-Validation>.
+--
+-- Here's the full schema:
+--
+-- @
+-- enum DogCommand { SIT, DOWN, HEEL }
+--
+-- type Dog implements Pet {
+--   name: String!
+--   nickname: String
+--   barkVolume: Int
+--   doesKnowCommand(dogCommand: DogCommand!): Boolean!
+--   isHousetrained(atOtherHomes: Boolean): Boolean!
+--   owner: Human
+-- }
+--
+-- interface Sentient {
+--   name: String!
+-- }
+--
+-- interface Pet {
+--   name: String!
+-- }
+--
+-- type Alien implements Sentient {
+--   name: String!
+--   homePlanet: String
+-- }
+--
+-- type Human implements Sentient {
+--   name: String!
+-- }
+--
+-- enum CatCommand { JUMP }
+--
+-- type Cat implements Pet {
+--   name: String!
+--   nickname: String
+--   doesKnowCommand(catCommand: CatCommand!): Boolean!
+--   meowVolume: Int
+-- }
+--
+-- union CatOrDog = Cat | Dog
+-- union DogOrHuman = Dog | Human
+-- union HumanOrAlien = Human | Alien
+-- @
+--
+-- Unlike the spec, we don't define a @QueryRoot@ type here, instead
+-- encouraging test modules to define their own as appropriate to their needs.
+--
+-- We'll repeat bits of the schema below, explaining how they translate into
+-- Haskell as we go.
+
+module ExampleSchema
+  ( DogCommand
+  , DogCommandEnum
```

Comment:
Agreed. Have updated the SchemaTests to just construct `Enum "DogCommand" DogCommand` themselves.